src/components/structures/FavouriteMessagesView.tsx

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2015, 2016 OpenMarket Ltd
+Copyright 2017 Vector Creations Ltd
+Copyright 2018, 2019 New Vector Ltd
+Copyright 2019 - 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { MatrixClient, MatrixEvent } from 'matrix-js-sdk/src/matrix';
+import React, { useContext, useRef } from 'react';
+
+import MatrixClientContext from '../../contexts/MatrixClientContext';
+import { _t } from '../../languageHandler';
+import FavouriteMessageTile from '../views/rooms/FavouriteMessageTile';
+import ScrollPanel from './ScrollPanel';
+
+function getFavouriteMessagesTiles(cli: MatrixClient, favouriteMessageEvents) {
```

Comment:
This should probably be a functional component

Reply:
Noted